openverse_catalog/dags/maintenance/flickr_audit_sub_provider_workflow.py

```diff
@@ -0,0 +1,168 @@
+"""
+# Flickr Sub Provider Audit
+
+Check the list of member institutions of the Flickr Commons for institutions that
+have cc-licensed images and are not already configured as sub-providers for the Flickr
+DAG. Report suggestions for new sub-providers to Slack.
+"""
+
+import logging
+
+from airflow import DAG
+from airflow.exceptions import AirflowSkipException
+from airflow.models import Variable
+from airflow.operators.python import PythonOperator
+from common.constants import DAG_DEFAULT_ARGS
+from common.loader import provider_details as prov
+from common.requester import DelayedRequester
+from common.slack import send_alert
+from providers.provider_api_scripts.flickr import LICENSE_INFO
+
+
+logger = logging.getLogger(__name__)
+
+DAG_ID = "flickr_audit_sub_provider_workflow"
+MAX_ACTIVE = 1
+
+
+class FlickrSubProviderAuditor:
+
+    endpoint = "https://www.flickr.com/services/rest"
+    retries = 2
+
+    # Institutions with fewer than this number of cc-licensed images will not be
+    # recommended as a new sub provider.
+    minimum_image_count = 300
```

Comment:
I think this is a reasonable count for now 👍 It will be interesting to see how this changes in the far future when we may have _lots of_ individual sources. I could see eliminating it entirely.